client/pages/BikeInfo.tsx

```diff
@@ -86,41 +98,112 @@ const BikeInfo = ({route, navigation}) => {
+                            <View style={styles.containerRowsMedium}>
+                                <Icon name='map-marker' size={iconSizes.md} style={[styles.itemRowSpaceRight, styles.iconGreen]} />
+                                <Text style={styles.textHightlightMedium}>
+                                    {distToBike(userLocation, bike.location)} meters
+                                </Text>
+                            </View>
+                            <Text style={styles.textHightlightMedium}>{bikeStatus}</Text>
+                            <View>
+                                <Rating stars={bike.agg_rating} maxStars={5} size={iconSizes.md} color={colors.green} />
+                            </View>
                         </View>
                     </View>
-                    <Text style={styles.textMedium}>{bike.description}</Text>
-                    
                 </View>
-            </ScrollView>
-            <View style={styles.buttonBottomContainer}>  
-                {!userProfile.waiver ? <Text style={[styles.centerAlignText, styles.centerJustify, styles.textSmall, styles.headerContainerSmall]}>You must sign the accident waiver before continuing!</Text> : <Text></Text>}       
-                <TouchableOpacity onPress={() => onStartTripButton()} style={!userProfile.waiver ? styles.buttonDisabled : styles.buttonBottom} disabled={!userProfile.waiver}>
-                    <Text style={styles.buttonBottomText}>Start Trip</Text>
-                </TouchableOpacity>
+                <ScrollView style={[styles.containerFlex, styles.textScrollContainer]}>
+                        <Text style={styles.textMedium}>{bike.description}</Text>
+                </ScrollView>
+            </View>
+
+            {!userProfile.waiver ? <Text style={[styles.centerAlignText, styles.centerJustify, styles.textSmall, styles.headerContainerSmall]}>You must sign the accident waiver before continuing!</Text> : null} 
+
+            <View style={[styles.containerRowsNoPadding, styles.buttonBottomContainer]}>
+                <View style={[styles.buttonBottom, styles.itemRowSpaceRight, !userProfile.waiver || bikeStatus !== 'available' ? styles.buttonGray : styles.buttonGreen]} >  
+                    <Pressable onPress={() => onStartTripButton()} disabled={!userProfile.waiver || bikeStatus !== 'available'}>
+                        <Text style={styles.buttonBottomText}>Start Trip</Text>
+                    </Pressable>
+                </View>
+                <View style={[styles.buttonBottom, styles.itemRowSpaceLeft, !userProfile.waiver || bikeStatus !== 'available' ? styles.buttonGray : styles.buttonRed]}> 
+                    <Modal
+                        animationType="slide"
+                        transparent={true}
+                        visible={modalVisible}
+                        onRequestClose={() => {setModalVisible(!modalVisible);}}
+                        onDismiss={handleReportDamages}
```

Comment:
Looking at the docs: https://reactnative.dev/docs/next/modal#ondismiss-ios, onDismiss is only available on iOS, so on android, when I click the Submit Issue button, the handleReportDamages function is never triggered. Therefore, the status of the bike never gets updated to missing or broken. Is there a more universal way of doing this that works for both platforms?

Reply:
Thanks for catching that! I think we can just use an onPress with the close modal button instead of onDismiss